[Meng Han] iP

mhchan163 opened the pull request.

mhchan163 committed:
Added Level 0.Greet

mhchan163 committed:
Added Duke level 1. Greet, Echo, Exit

mhchan163 committed:
Added Duke Level 2.Add, List

mhchan163 committed:
Duke Level 3.Mark as Done

mhchan163 committed:
Duke level 4.ToDos, Events, Deadlines

ShawnTanzc commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+public class Event extends Task{
+    protected String by;
+
+    public Event(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (by: " + by + ")";
+    }
+    public String getStatusIcon() {
+        return super.getStatusIcon();
+    }
```
Could there be an inconsistency with the spacing?

ShawnTanzc commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,14 @@
+public class ToDo extends Task{
+
+    public ToDo(String description) {
+        super(description);
+    }
+
+    public String toString(){
+        return "[T]" + super.toString();
+    }
+    public String getStatusIcon() {
+       return super.getStatusIcon();
+    }
```
I think there could be a similar

ShawnTanzc reviewed (commented):
Good job otherwise! Keep up the good work

ivanderjmw commented on src/main/java/Duke.java:
```diff
@@ -17,28 +17,42 @@ public static void main(String[] args) {
+                list[count++] = new ToDo(line.substring(4));
                 System.out.println("____________________________________________________________");
-                System.out.println("Added: "+ line);
+                System.out.println("Got it. I've added this task:"+System.lineSeparator()+list[count-1].toString());
+                System.out.println("Now you have " + count + " tasks in your list.");
+                System.out.println("____________________________________________________________");
+            } else if(line.contains("deadline")){
+                String[] temp = line.split("/");
+                list[count++] = new Deadline(temp[0].substring(8),temp[1]);
+                System.out.println("____________________________________________________________");
+                System.out.println("Got it. I've added this task:" +System.lineSeparator()+list[count-1].toString());
+                System.out.println("Now you have " + count + " tasks in your list.");
+                System.out.println("____________________________________________________________");
+            } else if(line.contains("event")){
+                String[] temp = line.split("/");
+                list[count++] = new Event(temp[0].substring(5),temp[1]);
+                System.out.println("____________________________________________________________");
+                System.out.println("Got it. I've added this task:" +System.lineSeparator()+list[count-1].toString());
+                System.out.println("Now you have " + count + " tasks in your list.");
```
The code seems a bit repetitive here. It would be great if you modify this part to be less repetitive. The lines here also seem to be a bit long.

ivanderjmw replied:
The code seems a bit repetitive here. It would be great if you modify this part to be less repetitive. The lines here also seem to be a bit long.

ivanderjmw reviewed (commented):
Overall, I think your code is readable, however some parts could be simplified. Otherwise, LGTM!

Feudalord commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,16 @@
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
+    }
+    public String getStatusIcon() {
```
Redundant overide that doesn't change super class method. Consider removing the override to shorten the code.

Feudalord commented on src/main/java/Duke.java:
```diff
@@ -17,28 +17,42 @@ public static void main(String[] args) {
+                System.out.println("Now you have " + count + " tasks in your list.");
+                System.out.println("____________________________________________________________");
+            } else if(line.contains("deadline")){
+                String[] temp = line.split("/");
```
Nice spilt method to split string into 2 seperated by '/'

mhchan163 committed:
Added Duke level 5
-added printIntro method

mhchan163 committed:
Signed-off-by: mhchan163 <mhchan163@gmail.com>

mhchan163 committed:
Added Improvement in Coding Standards:
Specifically on exception classes(change the first word of the class to Capital letters in accordance to standard coding standards)

mhchan163 committed:
Empty commit

mhchan163 committed:
Added Duke level 6. Delete
Added A-Collections, converted Task Array to ArrayList

mhchan163 committed:
Added FileReading and File creation

mhchan163 committed:
Merge branch 'Branch-level-6' into Branch-level-7

# Conflicts:
#	src/main/java/Duke.java

Resolved Conflicts

mhchan163 committed:
Added Duke Level 7
Code is untidy and will be edited later

mhchan163 committed:
empty commit for jar

0WN463 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,139 @@
+        Scanner in = new Scanner(System.in);
+        ArrayList<Task> list = new ArrayList<>();
+        int count=0;
+        //Task[] list = new Task[100];
```
Avoid leaving commented code in your codebase. If you require this fragment in the future, you can retrieve it using version control. The danger of leaving it in is that it may be left there permanently

0WN463 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,139 @@
+        int count=0;
+        //Task[] list = new Task[100];
+        String line;
+        File f = new File("data/Duke.txt");
```
Single letter variables are reserved for iteration.
Consider using a more descriptive variable name.

0WN463 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,139 @@
+
+    public static void main(String[] args) throws UnknownCommandException,EmptyInputException,java.io.IOException{
+        printIntro();
+        Scanner in = new Scanner(System.in);
```
Consider using a more descriptive variable name

0WN463 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,139 @@
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void main(String[] args) throws UnknownCommandException,EmptyInputException,java.io.IOException{
```
It is a bit strange that your main is allowed to throw errors.
Shouldn't all errors be caught and handled by main ?

0WN463 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,139 @@
+                        list.get(count-1).markAsDone();
+                    }
+                } else if(temp[0].equals("[D] ")){
+                    list.add(new Deadline(temp[2], temp[3]));
```
Consider using temporary variables as labels so that others can understand what each fragment means

0WN463 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,139 @@
+        String line;
+        File f = new File("data/Duke.txt");
+        if(f.exists()){
+            Scanner s = new Scanner(f);
```
Consider using a more descriptive variable name.

0WN463 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,139 @@
+        } else {
+            f.createNewFile();
+        }
+        FileWriter fw = new FileWriter(f);
```
Consider using a more descriptive variable name.

0WN463 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,139 @@
+                    list.remove(index-1);
+                    count--;
+                }else if (line.startsWith("todo")) {
+                    list.add(new ToDo(line.substring(4)));
```
Avoid using magic numbers (4).
Try to redesign your logic so that it does not depends on such magic numbers

0WN463 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,139 @@
+                    }
+                    System.out.println("____________________________________________________________");
+                } else if (line.startsWith("done")) {
+                    String[] temp = line.split(" ");
```
Consider using a more descriptive variable name.
```suggestion
                    String[] stringFragments = line.split(" ");
```

0WN463 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,22 @@
+public class Event extends Task{
+    protected String by;
```
Is there a reason why this is protected instead of private?

0WN463 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+        this.isDone = false;
+    }
+
+    public String taskCode(){
+        return "";
+    }
+
+    public String getTime(){
+        //returns nothing here cause there is no time component
+        return "";
+    }
```
It seem strange that the taskcode is blank

And that getTime returns blank also. Perhaps you might need to tweak your hierarchy such that it each class only contains the fields it needs.

0WN463 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    public String toString() {
+        return getStatusIcon() +  description;
```
Inconsistent spacing

0WN463 reviewed (commented):
Overall seem ok.

Just be more careful about coding standard violations, as those do add up overtime.
Also look out for those minor code quality issues.

mhchan163 committed:
Created and added methods
-printOutro : prints "bye" message
-uploadSavedData : uploads saved data into list
-saveListData : saves the data in current list into the save file
-processInputCommand : asks for input and executes the command

mhchan163 committed:
Added Storage but has pathfile erros

mhchan163 committed:
Added Ui, Storage, Parser and TaskList classes for MoreOOP

mhchan163 committed:
Added Incomplete LocalDate

mhchan163 committed:
Added Search function but has not been tested

mhchan163 committed:
Added Search function for Level-9

mhchan163 committed:
Added Level 8: Dates and Times
Have yet to handle exception

mhchan163 committed:
Removed unnecessary Classes FileReader and OutofBoundsIndexException

mhchan163 committed:
Added javadoc comments for all classes and some non-trival methods

mhchan163 committed:
Merge pull request #1 from mhchan163/Level-8

Level 8

mhchan163 committed:
Merge Conflict test

mhchan163 committed:
Merge branch 'master' into Branch-level-9

# Conflicts:
#	src/main/java/TaskList.java

mhchan163 committed:
Merge branch 'master' into A-JavaDoc

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Storage.java

mhchan163 committed:
Merge pull request #2 from mhchan163/Branch-level-9

Branch level 9

mhchan163 committed:
Merge branch 'master' into A-JavaDoc

mhchan163 committed:
Merge pull request #3 from mhchan163/A-JavaDoc

Added javadoc comments for all classes and some non-trival methods

mhchan163 committed:
Added User guide

mhchan163 committed:
Changed some formatting

mhchan163 committed:
Fixed some format errors

mhchan163 committed:
Added packages

mhchan163 committed:
Added command class and wrong date format exception

mhchan163 committed:
Removed / expection for date, made some error fixes in deletecommand

mhchan163 committed:
Added Ui implementations

mhchan163 committed:
Added meta-inf and data to gitignore

mhchan163 committed:
Added another exception and change how event manages the time input

mhchan163 committed:
Added the other half of the edited files from previous commit

mhchan163 committed:
Added Javadoc comments for all classes and some methods

mhchan163 committed:
Added create directory if "Duke.txt" does not yet exists.

mhchan163 committed:
Added User Guide

mhchan163 committed:
Merge with master
